src/lib/operators/jit_operator/operators/jit_read_tuples.cpp

```diff
@@ -107,13 +107,15 @@ void JitReadTuples::before_specialization(const Table& in_table, std::vector<boo
 
   if (in_table.chunk_count() == 0) return;
 
-  const auto& chunk = *in_table.get_chunk(ChunkID{0});
+  // TODO(all) Should we create a get_first_valid_chunk() function
+  //  returning the first non-nullpointer-chunk of the table?
+  const auto chunk = in_table.get_chunk(ChunkID{0});
   // Remove expressions that use a column where the first segment is not dictionary-encoded
   _value_id_expressions.erase(
       std::remove_if(_value_id_expressions.begin(), _value_id_expressions.end(),
                      [&](const JitValueIdExpression& value_id_expression) {
                        const auto column_id = _input_columns[value_id_expression.input_column_index].column_id;
-                       const auto segment = chunk.get_segment(column_id);
+                       const auto segment = chunk->get_segment(column_id);
```

Comment:
You added a todo, but this looks like it's currently broken if the first chunk is deleted.

Reply:
Ok, I removed the todo. Is there anything I should change?